helpers/deep_freeze_object.js

```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+module.exports = function deepFreeze(o) {
+	Object.freeze(o);
+
+	Object.getOwnPropertyNames(o).forEach(prop => {
+		if (
+			Object.prototype.hasOwnProperty.call(o, prop) &&
+			o[prop] !== null &&
+			(typeof o[prop] === 'object' || typeof o[prop] === 'function') &&
```

Comment:
I copied the logic from here: https://stackoverflow.com/a/34776962/1417536

I think they just wanted to follow what official documentation says: ` Note that strings, numbers, and booleans are always immutable and that Functions and Arrays are objects.`.  See: [Freezing Arrays](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze#Freezing_Arrays)

I think it doesn't bring any value, since the property which holds the function as value will be already frozen and there is no need to freeze a function. I looked into `new Fn()` usage but `typeof new Fn()` is an object. 

hasOwnProperty check is completely useless though. I'll remove that.